extensions/amp-pinterest/0.1/pinit-button.js

```diff
@@ -0,0 +1,187 @@
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {Util} from './util';
+
+// Popup options
+const POP = 'status=no,resizable=yes,scrollbars=yes,' +
+  'personalbar=no,directories=no,location=no,toolbar=no,' +
+  'menubar=no,width=900,height=500,left=0,top=0';
+
+/*
+ * Pinterest Pinit Button
+ * @attr data-url: the source url for the Pin
+ * @attr data-media: the url of the Pin image/media
+ * @attr data-description: the description of the Pin
+ *
+ * OPTIONAL
+ * @attr data-color: the button color from [red, white, gray]
+ * @attr data-count: the position of the Pin count from [beside, above]
+ * @attr data-lang:  the language of the button from [en, ja]
+ * @attr data-round: should the button be round (true if set)
+ * @attr data-tall:  should the button be tall  (true if set)
+ **/
+export class PinItButton {
+
+  constructor(rootElement) {
+    AMP.assert(rootElement.getAttribute('data-url'),
+      'The data-url attribute is required for Pin It buttons');
+    AMP.assert(rootElement.getAttribute('data-media'),
+      'The data-media attribute is required for Pin It buttons');
+    AMP.assert(rootElement.getAttribute('data-description'),
+      'The data-description attribute is required for Pin It buttons');
+    this.element = rootElement;
+  }
+
+  /*
+   * Override the default href click handling to log and open popup
+   * @param {Event} evt: the HTML event object
+   **/
+  handleClick(evt) {
+    evt.preventDefault();
+    window.open(this.href, '_pinit', POP);
+    Util.log('&type=button_pinit');
+  }
+
+  /*
+   * Fetch the remote Pin count for the source URL
+   * @param {Event} evt: the HTML event object
+   * @returns {Promise}
+   **/
+  fetchCount() {
+    const url = `https://widgets.pinterest.com/v1/urls/count.json?return_jsonp=false&url=${this.url}`;
```

Comment:
It seems that `this.url` has to be pushed through `encodeURIComponent`. I think it's a good idea to do so for all URL params.